### PR TITLE
Fixes site settings Autoplay value selector.

### PR DESCRIPTION
**Pull request by mkarolin**
Fixes brave/brave-browser#10049

Content settings types list in constants.js is meant to be kept in sync
with chrome/browser/ui/webui/site_settings_helper.cc. We do not patch
the enum in site_settings_helper.cc, but instead override functions that
manipulate it.

On js side we can try to override the behavior, but that may introduce
more patches than just patching the constants.

## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by mkarolin** [comment]
CI: Unrelated unit-test failure on Mac

File: patches/chrome-browser-resources-settings-site_settings-constants.js.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+diff --git a/chrome/browser/resources/settings/site_settings/constants.js b/chrome/browser/resources/settings/site_settings/constants.js
+index 0bc63a422d1d1ca468a1222dd699d5fa9bb65d26..89dca72c11bd2ee16af52bed19170999b843ac84 100644
+--- a/chrome/browser/resources/settings/site_settings/constants.js
++++ b/chrome/browser/resources/settings/site_settings/constants.js
+@@ -42,6 +42,7 @@ cr.define('settings', function() {
+     HID_DEVICES: 'hid-devices',
+     AR: 'ar',
+     VR: 'vr',
++    AUTOPLAY: 'autoplay',
```
Comment (bridiver) [inline patches/chrome-browser-resources-settings-site_settings-constants.js.patch:9]: cc @petemill

Comment (petemill) [reply patches/chrome-browser-resources-settings-site_settings-constants.js.patch:9]: I had a look at this with @mkarolin yesterday and we decided the patch was the only way without overriding every single component which uses it (it's looking like a lot of them). So after a few minutes of looking for a solution, this seems to be the cleanest for now.

**Comment by bsclifton** [comment]
Beat me to the merge @mkarolin 😂 

CI looks good! macOS tests failed due to https://github.com/brave/brave-browser/issues/6515 but everything else built / passed just fine

**Pull request merged**